### PR TITLE
Add solution

### PR DESCRIPTION
**Pull request by Resci**

### CODE REVIEW

File: src/main/java/core/basesyntax/Main.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    public static void main(String[] args) {
+        Random randomGenerator = new Random();
+        int random = randomGenerator.nextInt(100);
+        Figures[] figures = new Figures[random];
+
+        for (int i = 0; i < random; i++) {
+            figures[i] = Figures.getRandomFigure();
+        }
+
+        for (Figures some: figures) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Main.java:15]:
```suggestion
        for (Figures figure/element: figures) {
```

File: src/main/java/core/basesyntax/Figures.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    public Color getColor() {
+        return color;
+    }
+
+    public double getArea() {
+        return area;
+    }
+
+    public static Figures getRandomFigure() {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figures.java:22]: With this logic your abstract class depends on the impl and it's a bad practice, move this logic to separate class, e.g. "FigureGenerator". This way your code will follow single responsibility principle

File: src/main/java/core/basesyntax/Figures.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class Figures {
+    private static Random random = new Random();
+    double area;
+    Color color;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figures.java:8]: what about access modifiers, should it be default?

File: src/main/java/core/basesyntax/Color.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    private static Random random = new Random();
+    private String code;
+
+    Color(String code) {
+        this.code = code;
+    }
+
+    public static Color getRandom() {
+        switch (random.nextInt(8)) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Color.java:23]: 8 is a magic number, why not 7 or 9 ?

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Color.java (hunk 4 above)
Comment (Resci) [reply src/main/java/core/basesyntax/Color.java:23]: We have 8 different color in enum structure, so i used random.nextInt(8) which return [0; 8). It is better to change it to variable? If do that, may appeaar situation, when someone add the new color but forget add it to random generator

Comment (RomanDubovskyi) [reply src/main/java/core/basesyntax/Color.java]: so instead you can use enum length which will make code more readable and flexible 
https://stackoverflow.com/questions/17492079/how-to-get-number-of-possible-item-of-an-enum/17492102

File: src/main/java/core/basesyntax/figures/Figures.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.figures;
+
+public abstract class Figures {
+
+    protected double area;
+    protected Color color;
+
+    public abstract void displayInfo();
+
+    public abstract void draw();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/figures/Figures.java:10]: these methods better to move to interface and make this class implement it (interface - behaviour, abstract class state)

File: src/main/java/core/basesyntax/figures/FigureGenerator.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
+package core.basesyntax.figures;
+
+import java.util.Random;
+
+public class FigureGenerator {
+    private static Random random = new Random();
+
+    public static Figures getRandomFigure() {
+        switch (random.nextInt(5)) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/figures/FigureGenerator.java:9]: declare 5 as a constant, not it's magic number

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/figures/FigureGenerator.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.figures;
+
+import java.util.Random;
+
+public class FigureGenerator {
+    private static final int count = 5; //Number of inherited classes
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/figures/FigureGenerator.java:6]:
```suggestion
    private static final int FIGURES_AMOUNT / FIGURES_QUANTITY = 5; 
```
make code self description, constant naming - https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

File: src/main/java/core/basesyntax/figures/Figures.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.figures;
+
+public abstract class Figures implements Displayable, Drawable {
+
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/figures/Figures.java:4]: this empty line here is redundant

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]
Good job, almost there 👍

File: src/main/java/core/basesyntax/figures/Figures.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.figures;
+
+public abstract class Figures implements Displayable, Drawable {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/figures/Figures.java:3]:
```suggestion
public abstract class Figure implements Displayable, Drawable {
```

File: src/main/java/core/basesyntax/figures/IsoscelesTrapezoid.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.figures;
+
+public class IsoscelesTrapezoid extends Figures {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/figures/IsoscelesTrapezoid.java:3]: For future - it's better to have getters and setters as well, so you can change figure state after creation,

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**